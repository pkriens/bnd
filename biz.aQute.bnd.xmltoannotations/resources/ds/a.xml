<?xml version="1.0" encoding="UTF-8"?>
<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" 
    name="aaa"
    activate="activate"
    deactivate="deactivate"
    modified="modified"
    factory="factory"
    immediate = "true"
    configuration-policy="optional"
    >
   <implementation class="test.A"/>
   <service>
      <provide interface="test.x.Foo"/>
   </service>
   <reference bind="setBar" cardinality="0..1" interface="test.x.Bar" name="Bar" policy="dynamic" unbind="unsetBar" updated="updatedBar"/>
   <reference field="bar" cardinality="0..1" interface="test.x.Bar" name="Bar" policy="dynamic" unbind="unsetBar" updated="updatedBar"/>
   <property name="foo" type="String" value="bar"/>
   <property name="bar" type="String" value="foo"/>
   <property name="list" type="Integer">1
2
4
8
   </property>
</scr:component>
