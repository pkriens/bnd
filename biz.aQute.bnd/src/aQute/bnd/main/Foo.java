package foo.bar;

class Foo {
    /**
     * A function to turn a  Dictionary to a Map
     */
    public static Map<String, Object> toMap(Dictionary<String, Object> d) {
        Map<String, Object> map = new HashMap<String, Object>();
        for (Enumeration<String> e = d.keys(); e.hasMoreElements();) {
            String key = e.nextElement();
            map.put(key, d.get(key));
        }
        return map;
    }

    /**
     * Use bnd HttpClient to get a URL, convert the result body from JSON to a Map
     * 
     */

     //+ add a simple java class that implements a linked list with next and prev

        
}