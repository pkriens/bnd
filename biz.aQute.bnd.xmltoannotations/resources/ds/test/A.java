package test;

import test.x.Foo;
import test.x.Bar;

public class A implements Foo {
	
	Bar bar;
	
	void activate(BundleContext context) {
		
	}
	
	void deactivate() {
		
	}
	
	void modified() {
		
	}
	
	void setBar(Bar bar) {
		
	}
	
	void unsetBar(Bar bar) {
		
	}
	
	void updatedBar(Bar bar) {
		
	}
	
}