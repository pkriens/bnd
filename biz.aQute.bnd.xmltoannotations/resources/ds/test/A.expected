package test;

import test.x.Foo;
import test.x.Bar;

@org.osgi.service.component.annotations.Component(factory = "factory", immediate = true, service = test.x.Foo.class, name = "aaa", property = { "foo=bar", "bar=foo", "list:Integer=1", "list:Integer=2", "list:Integer=4", "list:Integer=8" })
public class A implements Foo {

    @org.osgi.service.component.annotations.Reference(unbind = "unsetBar", service = test.x.Bar.class, name = "Bar", updated = "updatedBar", cardinality = org.osgi.service.component.annotations.ReferenceCardinality.OPTIONAL, policy = org.osgi.service.component.annotations.ReferencePolicy.DYNAMIC)
    Bar bar;

    @org.osgi.service.component.annotations.Activate
    void activate(BundleContext context) {
    }

    @org.osgi.service.component.annotations.Deactivate
    void deactivate() {
    }

    @org.osgi.service.component.annotations.Modified
    void modified() {
    }

    @org.osgi.service.component.annotations.Reference(unbind = "unsetBar", service = test.x.Bar.class, name = "Bar", updated = "updatedBar", cardinality = org.osgi.service.component.annotations.ReferenceCardinality.OPTIONAL, policy = org.osgi.service.component.annotations.ReferencePolicy.DYNAMIC)
    void setBar(Bar bar) {
    }

    void unsetBar(Bar bar) {
    }

    void updatedBar(Bar bar) {
    }
}
